### ISSUE TITLE
Dataset _getittem_ Iterator

### ISSUE THREAD

**Issue by tchaton** [pytorch/pytorch#14229]
Hello,

I was wondering Dataset could support in some way an iterator.

I need to generate from x (image), N image with a part masked in a sliding window way.
I cant't do it with the current format, or I need to externalize it and 
convert tensor -> numpy (modify it) -> tensor.

Best.

**Comment by fmassa** [comment]
You can do something like that with a double dataloader: the first one selects the images (as it is now), and the second one can take the images and return the sliding windows (or a subset of them).

How does that sound?

**Comment by SsnL** [comment]
A dataset in the form of an iterable is in the works. I will finish it
around early December.

**Comment by SsnL** [comment]
implemented now!
